packages/server-core/src/media/recursive-archiver/archiver.class.ts

```diff
@@ -24,87 +24,112 @@ Ethereal Engine. All Rights Reserved.
+  let result = await storageProvider.listFolderContent(directory)
 
-    logger.info(`Archiving ${directory} using ${storageProviderName}`)
+  const zip = new JSZip()
 
-    let result = await storageProvider.listFolderContent(directory)
+  for (let i = 0; i < result.length; i++) {
+    if (result[i].type == 'folder') {
+      let content = await storageProvider.listFolderContent(result[i].key + '/')
+      content.forEach((f) => {
+        result.push(f)
+      })
+    }
 
-    const zip = new JSZip()
+    if (result[i].type == 'folder') continue
 
-    for (let i = 0; i < result.length; i++) {
-      if (result[i].type == 'folder') {
-        let content = await storageProvider.listFolderContent(result[i].key + '/')
-        content.forEach((f) => {
-          result.push(f)
-        })
-      }
+    const blobPromise = await fetch(result[i].url, { method: 'GET' }).then((r) => {
+      if (r.status === 200) return r.arrayBuffer()
+      return Promise.reject(new Error(r.statusText))
+    })
 
-      if (result[i].type == 'folder') continue
+    logger.info(`Added ${result[i].key} to archive`)
 
-      const blobPromise = await fetch(result[i].url, { method: 'GET' }).then((r) => {
-        if (r.status === 200) return r.arrayBuffer()
-        return Promise.reject(new Error(r.statusText))
-      })
+    const dir = result[i].key.substring(result[i].key.indexOf('/') + 1)
+    zip.file(dir, blobPromise)
+  }
 
-      logger.info(`Added ${result[i].key} to archive`)
+  const generated = await zip.generateAsync({ type: 'blob', streamFiles: true })
 
-      const dir = result[i].key.substring(result[i].key.indexOf('/') + 1)
-      zip.file(dir, blobPromise)
-    }
+  const zipOutputDirectory = `temp${directory.substring(directory.lastIndexOf('/'))}.zip`
 
-    const generated = await zip.generateAsync({ type: 'blob', streamFiles: true })
+  logger.info(`Uploading ${zipOutputDirectory} to storage provider`)
 
-    const zipOutputDirectory = `temp${directory.substring(directory.lastIndexOf('/'))}.zip`
+  await storageProvider.putObject({
+    Key: zipOutputDirectory,
+    Body: Buffer.from(await generated.arrayBuffer()),
+    ContentType: 'archive/zip'
+  })
 
-    logger.info(`Uploading ${zipOutputDirectory} to storage provider`)
+  logger.info(`Archived ${directory} to ${zipOutputDirectory}`)
 
-    await storageProvider.putObject({
-      Key: zipOutputDirectory,
-      Body: Buffer.from(await generated.arrayBuffer()),
-      ContentType: 'archive/zip'
-    })
+  return zipOutputDirectory
+}
 
-    logger.info(`Archived ${directory} to ${zipOutputDirectory}`)
+export class Archiver implements Partial<ServiceMethods<any>> {
+  app: Application
 
-    return zipOutputDirectory
+  constructor(app: Application) {
+    this.app = app
+  }
+
+  async setup(app: Application, path: string) {}
+
+  async get(directory: string, params?: ArchiveParams): Promise<string> {
+    if (!config.kubernetes.enabled || params?.isJob) return archive(directory, params)
```

Comment:
Having `!config.kubernetes.enabled` means for local development we wouldnt be using Git LFS. Is it because of assumption that in local environment file will be uploaded to local file server?

I am just wandering if this will impact once we move to MinIO?

Reply:
Unless I'm missing something, Git LFS is being used in all situations. I converted archiver, project-github-push, and project.update to either make a job or run the function they originally did. Locally, we never want to make a job, so those functions will just call their original logic, which is now in a separate function. That separate function, or sub-functions of it, will use LFS. In a deployed environment, the default is to make an executor job for it, with the job calling the function with isJob: true, which will run the intended function.